libraries/node-core-library/src/FileSystem.ts

```diff
@@ -505,6 +744,13 @@ export class FileSystem {
     fsx.symlinkSync(options.linkTargetPath, options.newLinkPath, 'dir');
   }
 
+  /**
+   * An async version of {@link FileSystem.createSymbolicLinkFolder}.
+   */
+  public static createSymbolicLinkFolderAsync(options: IFileSystemCreateLinkOptions): Promise<void> {
+    return LegacyAdapters.convertCallbackToPromise(fsx.symlink, options.linkTargetPath, options.newLinkPath, 'dir');
```

Comment:
I noticed that Node 10 introduces a built-in `promises` adapter for all the filesystem APIs (but it doesn't work in Node 8, which some of our projects still use):
https://nodejs.org/docs/latest-v10.x/api/fs.html#fs_fs_promises_api


Also `fs-extra` provides promise adapters for each API, based on [universalify](https://www.npmjs.com/package/universalify):

[node-fs-extra/lib/fs/index.js](https://github.com/jprichardson/node-fs-extra/blob/934ea759ad8f36121ccbabc2759cf033bce67f81/lib/fs/index.js#L59)
```ts
const u = require('universalify').fromCallback
const fs = require('graceful-fs')

. . .

// Export all keys:
Object.keys(fs).forEach(key => {
  if (key === 'promises') {
    // fs.promises is a getter property that triggers ExperimentalWarning
    // Don't re-export it here, the getter is defined in "lib/index.js"
    return
  }
  exports[key] = fs[key]
})

// Universalify async methods:
api.forEach(method => {
  exports[method] = u(fs[method])
})
```

It might be useful to add a code comment explaining what is the advantage of using `LegacyAdapters.convertCallbackToPromise()`.  (Or maybe consider one of those options.)

The decision factors I'd be most interested in are:
- How clean are the debugger callstacks?  Do the wrappers add annoying extra callstack records?
- How accurate are the TypeScript types?
- Does it work on Node.js 8?
- How good is the error handling?

Reply:
I changed direction to use the `fs-extra`-built-in promises functions. The callstacks are clean and the error messages seem to be just the node `fs` error messages. I updated the TypeScript `@types` package to fix some typings issues.

Everything I tested worked fine in Node 8.